src/magic.cpp

```diff
@@ -13657,10 +13808,17 @@ int cast_solidity( ubyte level, CHAR_DATA *ch, char *arg,
 	 break;
   case SPELL_TYPE_WAND:
   case SPELL_TYPE_SCROLL:
-      if (tar_obj) return eFAILURE;
-      if (!tar_ch) tar_ch = ch;
+       if(tar_obj)
+          return eFAILURE;
+       if(!tar_ch) tar_ch = ch;
+          return spell_solidity(level, ch, tar_ch, 0, skill);
+       break;
     return spell_solidity(level, ch, tar_ch, 0, skill);
```

Comment:
Your changes in this section should be undone,  The spacing is wrong, and confusing, and you aren't actually changing the behavior.  This also now orphans the final return line.